test/e2e/beat/recipes_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beat
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"testing"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	beatcommon "github.com/elastic/cloud-on-k8s/pkg/controller/beat/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/settings"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/kibana"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/beat"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test/helper"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+func TestFilebeatNoAutodiscoverRecipe(t *testing.T) {
+	name := "fb-no-autodiscover"
+	pod, loggedString := loggingTestPod(name)
+	customize := func(builder beat.Builder) beat.Builder {
+		return builder.
+			WithRoles(beat.PSPClusterRoleName).
+			WithESValidations(
+				beat.HasMessageContaining(loggedString),
+			)
+	}
+
+	runBeatRecipe(t, "filebeat_no_autodiscover.yaml", customize, pod)
+}
+
+func TestFilebeatAutodiscoverRecipe(t *testing.T) {
+	name := "fb-autodiscover"
+	pod, loggedString := loggingTestPod(name)
+	customize := func(builder beat.Builder) beat.Builder {
+		return builder.
+			WithRoles(beat.PSPClusterRoleName).
+			WithESValidations(
+				beat.HasEventFromPod(pod.Name),
+				beat.HasMessageContaining(loggedString),
+			)
+	}
+
+	runBeatRecipe(t, "filebeat_autodiscover.yaml", customize, pod)
+}
+
+func TestFilebeatAutodiscoverByMetadataRecipe(t *testing.T) {
+	name := "fb-autodiscover-meta"
+	podBad, badLog := loggingTestPod(name + "-bad")
+	podLabel, goodLog := loggingTestPod(name + "-label")
+	podLabel.Labels["log-label"] = "true"
+
+	customize := func(builder beat.Builder) beat.Builder {
+		return builder.
+			WithRoles(beat.PSPClusterRoleName, beat.AutodiscoverClusterRoleName).
+			WithESValidations(
```

Comment:
I think it's a bit odd that we define test validations as part of the builder, that builds the test subjects, but I guess this is a general problem in our e2e setup at the moment.